sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuTransitionOverrides.scala

```diff
@@ -428,6 +430,25 @@ class GpuTransitionOverrides extends Rule[SparkPlan] {
     plan.children.foreach(assertIsOnTheGpu(_, conf))
   }
 
+  /**
+   * This is intended for testing only and this only supports looking for an exec once.
+   */
+  private def validateExecsInGpuPlan(plan: SparkPlan, conf: RapidsConf): Unit = {
```

Comment:
Nit: I actually expected this to be in `GpuOverrides` rather than here, which makes me wonder if we have enough stuff to warrant a `GpuOverrideUtils`.  Not must-fix, just an observation that doesn't have to be handled in this PR.

Reply:
yeah I assume we would want to move assertIsOnTheGpu and related things there. I can do in followup.